### ISSUE TITLE
Hint mechanism to require more "slots" to build a crate

### ISSUE THREAD

**Issue by joshtriplett** [rust-lang/cargo#8405]
When building rust-lang/rust in parallel, I found that `rustc_middle` would take the vast majority of the compilation time and memory. I repeatedly ran into the OOM-killer because cargo would start up three more things building in parallel, and `rustc_middle` was already using 5-6GB of my 8GB of RAM. I'd like to be able to somehow hint to Cargo that it should treat `rustc_middle` as particularly heavy to build, and that it should count as multiple "slots" in the 4 parallel jobs Cargo would allow to run. That would have allowed me to successfully build within 8GB of RAM.

**Comment by luser** [comment]
Possibly a duplicate of #7437?

**Comment by epage** [comment]
#7437 is about package prioritization while this is about resource utilization.

**Comment by epage** [comment]
#12912 discusses the OOM problem and is exploring smaller and bigger solutions.  To keep the conversation in one place, I'm closing this.
